src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
```

Comment:
[NAMING] - The name 'currentState' is a bit generic. Consider renaming it to 'stateCopy' or something similar to indicate that it is a copy of the state object, which aligns with the good naming practices from the checklist.